tests/unit_tests/dft/source/compute_tests.cpp

```diff
@@ -39,244 +39,74 @@ extern std::vector<sycl::device *> devices;
-}
-
-TEST_P(ComputeTests, ComplexDoublePrecisionNotInPlaceRealRealUSM) {
-    auto test = DFT_Test<oneapi::mkl::dft::precision::DOUBLE, oneapi::mkl::dft::domain::COMPLEX>{
-        std::get<0>(GetParam()), std::get<1>(GetParam())
-    };
-    EXPECT_TRUEORSKIP(test.test_out_of_place_real_real_USM());
-}
-
-INSTANTIATE_TEST_SUITE_P(ComputeTestSuite, ComputeTests,
-                         ::testing::Combine(testing::ValuesIn(devices), testing::ValuesIn(lengths)),
-                         ::DimensionsDeviceNamePrint());
+class ComputeTests_in_place
+        : public ::testing::TestWithParam<std::tuple<sycl::device *, DFTParams>> {};
+class ComputeTests_real_real_in_place
+        : public ::testing::TestWithParam<std::tuple<sycl::device *, DFTParams>> {};
+class ComputeTests_out_of_place
+        : public ::testing::TestWithParam<std::tuple<sycl::device *, DFTParams>> {};
+class ComputeTests_real_real_out_of_place
+        : public ::testing::TestWithParam<std::tuple<sycl::device *, DFTParams>> {};
+
+#define INSTANTIATE_TEST(PRECISION, DOMAIN, PLACE, LAYOUT, STORAGE)                              \
+    TEST_P(ComputeTests##_##LAYOUT##PLACE, DOMAIN##_##PRECISION##_##PLACE##_##LAYOUT##STORAGE) { \
+        auto test =                                                                              \
+            DFT_Test<oneapi::mkl::dft::precision::PRECISION, oneapi::mkl::dft::domain::DOMAIN>{  \
+                std::get<0>(GetParam()), std::get<1>(GetParam()).sizes,                          \
+                std::get<1>(GetParam()).batches                                                  \
+            };                                                                                   \
+        EXPECT_TRUEORSKIP(test.test_##PLACE##_##LAYOUT##STORAGE());                              \
+    }
+
+#define INSTANTIATE_TEST_DIMENSIONS_PRECISION_DOMAIN(PLACE, LAYOUT, STORAGE) \
+    INSTANTIATE_TEST(SINGLE, COMPLEX, PLACE, LAYOUT, STORAGE)                \
+    INSTANTIATE_TEST(SINGLE, REAL, PLACE, LAYOUT, STORAGE)                   \
+    INSTANTIATE_TEST(DOUBLE, COMPLEX, PLACE, LAYOUT, STORAGE)                \
+    INSTANTIATE_TEST(DOUBLE, REAL, PLACE, LAYOUT, STORAGE)
+
+#define INSTANTIATE_TEST_DIMENSIONS_PRECISION_DOMAIN_PLACE_LAYOUT(STORAGE)      \
+    INSTANTIATE_TEST_DIMENSIONS_PRECISION_DOMAIN(in_place, , STORAGE)           \
+    INSTANTIATE_TEST_DIMENSIONS_PRECISION_DOMAIN(in_place, real_real_, STORAGE) \
+    INSTANTIATE_TEST_DIMENSIONS_PRECISION_DOMAIN(out_of_place, , STORAGE)       \
+    INSTANTIATE_TEST_DIMENSIONS_PRECISION_DOMAIN(out_of_place, real_real_, STORAGE)
+
+INSTANTIATE_TEST_DIMENSIONS_PRECISION_DOMAIN_PLACE_LAYOUT(buffer)
+INSTANTIATE_TEST_DIMENSIONS_PRECISION_DOMAIN_PLACE_LAYOUT(USM)
+
+using shape = std::vector<int64_t>;
+using i64 = std::int64_t;
+// Parameter format - { shape of transform, number of transforms }
+// due to how we are testing, the last dimension must be even
```

Comment:
That's not ideal, what is the issue exactly with odd sizes?

Reply:
addressed in https://github.com/oneapi-src/oneMKL/pull/275/commits/41f72310ccc0e6e81661d12cf658d0b8f7c3243a